sentry-android-core/src/main/java/io/sentry/android/core/internal/gestures/SentryGestureListener.java

```diff
@@ -178,12 +191,128 @@ private void addBreadcrumb(
 
     hub.addBreadcrumb(
         Breadcrumb.userInteraction(
-            eventType, ViewUtils.getResourceId(target), className, additionalData),
+            eventType, ViewUtils.getResourceIdWithFallback(target), className, additionalData),
         hintMap);
   }
 
+  private void startTracing(final @NotNull View target, final @NotNull String eventType) {
+    if (!(options.isTracingEnabled() && options.isEnableUserInteractionTracing())) {
+      return;
+    }
+
+    final Activity activity = activityRef.get();
+    if (activity == null) {
+      options.getLogger().log(SentryLevel.DEBUG, "Activity is null, no transaction captured.");
+      return;
+    }
+
+    final String viewId;
+    try {
+      viewId = ViewUtils.getResourceId(target);
+    } catch (Resources.NotFoundException e) {
+      options
+          .getLogger()
+          .log(
+              SentryLevel.DEBUG,
+              "View id cannot be retrieved from Resources, no transaction captured.");
+      return;
+    }
+
+    final View view = activeView.get();
+    if (activeTransaction != null) {
+      if (target.equals(view)
+          && eventType.equals(activeEventType)
+          && !activeTransaction.isFinished()) {
+        options
+            .getLogger()
+            .log(
+                SentryLevel.DEBUG,
+                "The view with id: "
+                    + viewId
+                    + " already has an ongoing transaction assigned. Rescheduling finish");
+
+        final Long idleTimeout = options.getIdleTimeout();
+        if (idleTimeout != null) {
```

Comment:
Here perhaps we'll add a check to do this only for scrolls/swipes

Reply:
We've decided to not distinguish between clicks and scrolls/swipes, so we'll just extend the transaction time for any kind of user interaction for the sake of simplicity